packages/twenty-front/src/modules/analytics/components/AnalyticsActivityGraph.tsx

```diff
@@ -33,34 +29,39 @@ const StyledTitleContainer = styled.div`
   justify-content: space-between;
 `;
 
-type SettingsDevelopersWebhookUsageGraphProps = {
-  webhookId: string;
-};
-
-export const SettingsDevelopersWebhookUsageGraph = ({
-  webhookId,
-}: SettingsDevelopersWebhookUsageGraphProps) => {
-  const webhookGraphData = useRecoilValue(webhookGraphDataState);
-  const setWebhookGraphData = useSetRecoilState(webhookGraphDataState);
+export const AnalyticsActivityGraph = ({
+  recordId,
+  endpointName,
+}: AnalyticsActivityGraphProps) => {
+  const [analyticsGraphData, setAnalyticsGraphData] = useRecoilComponentStateV2(
+    analyticsGraphDataComponentState,
+  );
   const theme = useTheme();
 
   const [windowLengthGraphOption, setWindowLengthGraphOption] = useState<
     '7D' | '1D' | '12H' | '4H'
   >('7D');
 
-  const { fetchGraphData } = useGraphData(webhookId);
+  const { fetchGraphData } = useGraphData({
+    recordId,
+    endpointName,
+  });
+
+  const transformDataFunction = computeAnalyticsGraphDataFunction(endpointName);
 
+  const dropdownId = useId();
+  // perhaps here i need to separate the Section container and the graph itself? TODO: Add elements of distintion btwen graphs of the same record type
   return (
     <>
-      {webhookGraphData.length ? (
+      {analyticsGraphData.length ? (
         <Section>
           <StyledTitleContainer>
             <H2Title
-              title="Activity"
-              description="See your webhook activity over time"
+              title={`${ANALYTICS_GRAPH_TITLE_MAP[endpointName]}`}
+              description={`See your ${ANALYTICS_ENDPOINT_TYPE_MAP[endpointName]} activity over time`}
```

Comment:
Let's stay consistent and use a map for descriptions, just like you did for title. This will give us more flexibility to customize the text. (note we could have also passed title/description as parameter of the component, that's how I would have done it intuitively but the way you did it works too!)